### PR TITLE
[benchmark] Update 'compare' printed ratios, compute geometric mean

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: hail/python/hailtop/hailctl/dev/benchmark/compare/compare.py [hunk 1]
```diff
@@ -43,27 +40,45 @@ def compare(run1, run2):
...
     for name, r1, r2 in comparison:
         comps.append(r2 / r1)
-        print(f'{name:>{longest_name}}   {fmt_diff(r2 / r1):>8}   {fmt_time(r1, 7):>7}   {fmt_time(r2, 7):>7}')
+        print(format(name, fmt_diff(r2 / r1), fmt_time(r1, 7), fmt_time(r2, 7)))
 
     print('----------------')
-    print(f'Average: {fmt_diff(np.mean(comps))}')
-    print(f'Median:  {fmt_diff(np.median(comps))}')
\ No newline at end of file
+    print(f'Geometric mean: {fmt_diff(np.sum(comps) ** (1 / len(comps)))}')
```
Comment (patrick-schultz) [inline hail/python/hailtop/hailctl/dev/benchmark/compare/compare.py]: should be product of the `comps`

File: hail/python/hailtop/hailctl/dev/benchmark/compare/compare.py [hunk 2]
```diff
@@ -18,15 +18,12 @@ def load_file(path):
...
-    return max(time1 / time2, time2 / time1)
-
-
 def fmt_diff(ratio):
     if ratio < 1:
-        return f'+{1 / ratio:.3f}'
+        return f'+{(1 / ratio - 1):.3f}'
     else:
-        return f'-{ratio:.3f}'
+        return f'-{(ratio - 1):.3f}'
```
Comment (patrick-schultz) [inline hail/python/hailtop/hailctl/dev/benchmark/compare/compare.py]: Personally, I find this confusing, and I have to keep running through in my head what positive vs. negative numbers mean. I think just reporting `ratio` (maybe as a percentage) is perfectly clear. But if you like this better I'm okay with it.

File: hail/python/hailtop/hailctl/dev/benchmark/compare/compare.py (hunk 1 above)
Comment (tpoterba) [reply hail/python/hailtop/hailctl/dev/benchmark/compare/compare.py]: whoops

File: hail/python/hailtop/hailctl/dev/benchmark/compare/compare.py (hunk 2 above)
Comment (tpoterba) [reply hail/python/hailtop/hailctl/dev/benchmark/compare/compare.py]: yeah, I think you're right.

**Pull request merged**